### PR TITLE
[DEVOPS-810] Move Safari build to browser workflow

### PR DESCRIPTION
**Pull request by michalchecinski**
## Type of change

- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [x] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Move the Safari extension build task from the desktop build workflow to the browser build.

Use the build Safari extension artifact in the desktop build to prepare mac OS packages

## Before you submit

- [ ] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Comment by michalchecinski** [comment]
Runs using changed workflow:
Browser: https://github.com/bitwarden/clients/actions/runs/2314180863
Desktop: https://github.com/bitwarden/clients/actions/runs/2320520008

File: .github/workflows/build-desktop.yml [hunk 1]
```diff
@@ -946,17 +976,34 @@ jobs:
...
+          branch: rc
+          path: ${{ github.workspace }}/browser-build-artifacts
+
+      - name: Download artifact from master
+        if: github.ref != 'refs/heads/rc'
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: build-browser.yml
+          workflow_conclusion: success
+          branch: DEVOPS-810_safari_build_browser # master
```
Comment (mimartin12) [inline .github/workflows/build-desktop.yml:994]:
```suggestion
          branch: master
```

File: .github/workflows/build-desktop.yml [hunk 2]
```diff
@@ -608,17 +602,36 @@ jobs:
         if: steps.build-cache.outputs.cache-hit != 'true'
         run: npm run build
 
-      - name: Build Safari extension
-        if: steps.safari-cache.outputs.cache-hit != 'true'
+      - name: Download artifact from rc
+        if: github.ref == 'refs/heads/rc'
+        uses: dawidd6/action-download-artifact@v2
```
Comment (joseph-flinn) [inline .github/workflows/build-desktop.yml:607]:
```suggestion
        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2.19.0
```

File: .github/workflows/build-desktop.yml [hunk 3]
```diff
@@ -608,17 +602,36 @@ jobs:
...
+        with:
+          workflow: build-browser.yml
+          workflow_conclusion: success
+          branch: rc
+
+          path: ${{ github.workspace }}/browser-build-artifacts
+
+      - name: Download artifact from master
+        if: github.ref != 'refs/heads/rc'
+        uses: dawidd6/action-download-artifact@v2
```
Comment (joseph-flinn) [inline .github/workflows/build-desktop.yml:617]:
```suggestion
        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2.19.0
```

File: .github/workflows/build-desktop.yml [hunk 4]
```diff
@@ -781,17 +794,34 @@ jobs:
         if: steps.build-cache.outputs.cache-hit != 'true'
         run: npm run build
 
-      - name: Build Safari extension
-        if: steps.safari-cache.outputs.cache-hit != 'true'
+      - name: Download artifact from rc
+        if: github.ref == 'refs/heads/rc'
+        uses: dawidd6/action-download-artifact@v2
```
Comment (joseph-flinn) [inline .github/workflows/build-desktop.yml:799]:
```suggestion
        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2.19.0
```

File: .github/workflows/build-desktop.yml [hunk 5]
```diff
@@ -781,17 +794,34 @@ jobs:
...
+        with:
+          workflow: build-browser.yml
+          workflow_conclusion: success
+          branch: rc
+          path: ${{ github.workspace }}/browser-build-artifacts
+
+      - name: Download artifact from master
+        if: github.ref != 'refs/heads/rc'
+        uses: dawidd6/action-download-artifact@v2
```
Comment (joseph-flinn) [inline .github/workflows/build-desktop.yml:808]:
```suggestion
        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2.19.0
```

File: .github/workflows/build-desktop.yml [hunk 6]
```diff
@@ -946,17 +976,34 @@ jobs:
         if: steps.build-cache.outputs.cache-hit != 'true'
         run: npm run build
 
-      - name: Build Safari extension
-        if: steps.safari-cache.outputs.cache-hit != 'true'
+      - name: Download artifact from rc
+        if: github.ref == 'refs/heads/rc'
+        uses: dawidd6/action-download-artifact@v2
```
Comment (joseph-flinn) [inline .github/workflows/build-desktop.yml:981]:
```suggestion
        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2.19.0
```

File: .github/workflows/build-desktop.yml [hunk 7]
```diff
@@ -946,17 +976,34 @@ jobs:
...
+        with:
+          workflow: build-browser.yml
+          workflow_conclusion: success
+          branch: rc
+          path: ${{ github.workspace }}/browser-build-artifacts
+
+      - name: Download artifact from master
+        if: github.ref != 'refs/heads/rc'
+        uses: dawidd6/action-download-artifact@v2
```
Comment (joseph-flinn) [inline .github/workflows/build-desktop.yml:990]:
```suggestion
        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2.19.0
```

**Review by joseph-flinn — CHANGES_REQUESTED** [review]
We've decided to pin the commits for the actions we use for both reliability (if the upstream action breaks in a new release, we aren't affected) and to help avoid supply chain attacks.

File: .github/workflows/build-desktop.yml (hunk 2 above)
Comment (michalchecinski) [reply .github/workflows/build-desktop.yml]: Done

File: .github/workflows/build-desktop.yml (hunk 3 above)
Comment (michalchecinski) [reply .github/workflows/build-desktop.yml]: Done

File: .github/workflows/build-desktop.yml (hunk 4 above)
Comment (michalchecinski) [reply .github/workflows/build-desktop.yml]: Done

File: .github/workflows/build-desktop.yml (hunk 5 above)
Comment (michalchecinski) [reply .github/workflows/build-desktop.yml]: Done

File: .github/workflows/build-desktop.yml (hunk 6 above)
Comment (michalchecinski) [reply .github/workflows/build-desktop.yml]: Done

File: .github/workflows/build-desktop.yml (hunk 7 above)
Comment (michalchecinski) [reply .github/workflows/build-desktop.yml]: Done

File: .github/workflows/build-desktop.yml (hunk 1 above)
Comment (michalchecinski) [reply .github/workflows/build-desktop.yml]: Done

File: .github/workflows/build-browser.yml [hunk 8]
```diff
@@ -222,6 +320,7 @@ jobs:
       - setup
       - locales-test
       - build
+      - build-safari
       - crowdin-push
     steps:
       - name: Check if any job failed
```
Comment (vgrassia) [inline .github/workflows/build-browser.yml:326]: In the step below you'll need to add an environment variable for the `build-safari` job status and add a check for failure in the `run` logic.

File: .github/workflows/build-desktop.yml [hunk 9]
```diff
@@ -608,17 +602,36 @@ jobs:
...
+
+      - name: Download artifact from master
+        if: github.ref != 'refs/heads/rc'
+        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2.19.0
+        with:
+          workflow: build-browser.yml
+          workflow_conclusion: success
+          branch: master
+
+          path: ${{ github.workspace }}/browser-build-artifacts
```
Comment (vgrassia) [inline .github/workflows/build-desktop.yml:623]:
```suggestion
          path: ${{ github.workspace }}/browser-build-artifacts
```

**Review by vgrassia — CHANGES_REQUESTED** [review]

File: .github/workflows/build-desktop.yml [hunk 10]
```diff
@@ -608,17 +602,36 @@ jobs:
...
-        if: steps.safari-cache.outputs.cache-hit != 'true'
+      - name: Download artifact from rc
+        if: github.ref == 'refs/heads/rc'
+        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2.19.0
+        with:
+          workflow: build-browser.yml
+          workflow_conclusion: success
+          branch: rc
+
+          path: ${{ github.workspace }}/browser-build-artifacts
```
[the remark this answers is not in the document]
Comment (michalchecinski) [reply .github/workflows/build-desktop.yml]: Done

File: .github/workflows/build-desktop.yml (hunk 9 above)
Comment (michalchecinski) [reply .github/workflows/build-desktop.yml]: Done

File: .github/workflows/build-browser.yml (hunk 8 above)
Comment (michalchecinski) [reply .github/workflows/build-browser.yml:326]: Done :)

**Review by vgrassia — APPROVED** [review]

**Pull request merged**